### PR TITLE
Battle detail

### PR DESCRIPTION
**Pull request by pantoja**
## Description
Display details of a battle in a view. It works for both active and inactive battles. Also, non-participants of a battle are not allowed to enter the page.

## Trello Card
As a Trainer I see the details of a settled battle
## Screenshots (if appropriate):
Inactive battle:
image
Active battle:
image

## Steps to reproduce (if appropriate):
1. Login with username: `admin@admin.com` and password `senhaadmin`
2. Navigate to [settled battles](https://pokebattle-battle-detai-aupfht.herokuapp.com/settled-battles/) and click in any battle to see the details of this battle
3. Do the same thing with [active battles](https://pokebattle-battle-detai-aupfht.herokuapp.com/active-battles/)
4. Try to enter [this page](https://pokebattle-battle-detai-aupfht.herokuapp.com/battle/3), you are not part of this battle so you should be redirected.
5. The same with [this page](https://pokebattle-battle-detai-aupfht.herokuapp.com/create-team/3), it's an invite to a battle, but you are not part of it too :disappointed:

### CODE REVIEW

File: backend/battles/tests/test_views.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
```
Comment (vanessa) [inline backend/battles/tests/test_views.py:3]: So proud of you writing tests 🌟

File: backend/battles/views.py [hunk 2]
```diff
@@ -86,3 +88,33 @@ def get_context_data(self, **kwargs):  # noqa
...
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        # Determine if you are opponent or creator
+        battle = context["battle"]
+        if self.request.user == battle.user_creator:
+            opponent = battle.user_opponent
+        else:
+            opponent = battle.user_creator
```
Comment (vanessa) [inline backend/battles/views.py:105]: How about you extract this to a helper?

File: backend/battles/views.py [hunk 3]
```diff
@@ -86,3 +88,33 @@ def get_context_data(self, **kwargs):  # noqa
...
+
+        context["opponent"] = opponent.get_short_name
```
Comment (vanessa) [inline backend/battles/views.py:107]: Is `get_short_name` a function or property? If it's a property, remove the `get_`, if it's a function, you need to call it.

File: backend/battles/views.py [hunk 4]
```diff
@@ -86,3 +88,33 @@ def get_context_data(self, **kwargs):  # noqa
...
+        # Battle is settled
+        if battle.settled:
+            context["winner"] = battle.winner.get_short_name
+            opponent_team = opponent.teams.get(battle=battle).team.all()
+            context["pokemon"] = zip(your_team, opponent_team)
+
+            return context
+
+        # Battle is not settled
+        context["pokemon"] = zip(your_team, [0, 0, 0])
```
Comment (vanessa) [inline backend/battles/views.py:119]: How about you extract this to a helper?

File: backend/templates/battles/detail_battle.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{% extends 'base.html' %}
+{% load static %}
+
+{% block title %}Your Battle{% endblock %}
+
+{% block body %}
+{% include "includes/page-title.html" with title=battle %}
```
Comment (vanessa) [inline backend/templates/battles/detail_battle.html:7]: NICE!

File: backend/templates/includes/pokemon-card.html [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<div class="pokemon-card">
```
Comment (vanessa) [inline backend/templates/includes/pokemon-card.html:1]: Very nice this componentization!

File: backend/battles/admin.py [hunk 7]
```diff
@@ -7,5 +7,9 @@ class TeamAdmin(admin.ModelAdmin):
     list_display = ("trainer", "battle")
```
Comment (vanessa) [inline backend/battles/admin.py:7]: image

Is the button supposed to be with this small offset?

Comment (vanessa) [inline backend/battles/admin.py:7]: image

Can I select the same pokemon as my opponent? I remember not being allowed on the specs.

Comment (pantoja) [reply backend/battles/admin.py:7]: I suppose so, there were no restriction of duplicate teams in the project specs.

Comment (pantoja) [reply backend/battles/admin.py:7]: Is that a frontend question?

File: backend/battles/views.py (hunk 4 above)
Comment (pantoja) [reply backend/battles/views.py:119]: Done! It's a difficult code to organize but I hope it's ok

File: backend/battles/views.py (hunk 2 above)
Comment (vanessa) [reply backend/battles/views.py:105]: I meant something like `get_opponent` where you would receive the battle and the `request.user` and return it.

File: backend/battles/views.py (hunk 3 above)
Comment (vanessa) [reply backend/battles/views.py:107]: Why did you mark it as resolved? Was it answered somewhere else?

File: backend/battles/views.py (hunk 4 above)
Comment (vanessa) [reply backend/battles/views.py:119]: I can't see the code (?), did you move it somewhere else?

**Pull request merged**